### DISCUSSION TITLE
Best practice to implement time based trades?

### DISCUSSION THREAD

**Discussion by seven7e** [kernc/backtesting.py#1151] [Q&A]
Say I'd like to hold a position for up to 5 days after its entry. What's the best way to do it?

**Comment by s-kust** [comment]
You may find this repo - [https://github.com/s-kust/python-backtesting-template/ ](https://github.com/s-kust/python-backtesting-template/ )- helpful

See the `process_max_duration` function in file [https://github.com/s-kust/python-backtesting-template/blob/main/utils/strategy_exec/special_situations.py](https://github.com/s-kust/python-backtesting-template/blob/main/utils/strategy_exec/special_situations.py)

**Comment by iimka** [comment]
You can check the [post](https://github.com/kernc/backtesting.py/discussions/869); I hope it helps you.
